src/Features/CSharp/Portable/AddAccessibilityModifiers/CSharpAddAccessibilityModifiersDiagnosticAnalyzer.cs

```diff
@@ -131,12 +133,32 @@ public CSharpAddAccessibilityModifiersDiagnosticAnalyzer()
 
             // Have an issue to flag, either add or remove. Report issue to user.
             var additionalLocations = ImmutableArray.Create(member.GetLocation());
+
+            var properties = ImmutableDictionary.CreateBuilder<string, string>();
+            var ruleName = CodeStyleOptions.RequireAccessibilityModifiers.StorageLocations.OfType<EditorConfigStorageLocation<CodeStyleOption<AccessibilityModifiersRequired>>>().FirstOrDefault();
+            if (ruleName != null)
+            {
+                properties["OptionName"] = ruleName.KeyName;
+                if (option.Value == AccessibilityModifiersRequired.ForNonInterfaceMembers)
+                {
+                    properties["OptionCurrent"] = "for_non_interface_members";
+                }
+                else if (option.Value == AccessibilityModifiersRequired.OmitIfDefault)
+                {
+                    properties["OptionCurrent"] = "omit_if_default";
+                }
+                else
+                {
+                    properties["OptionCurrent"] = option.Value.ToString().ToLowerInvariant();
```

Comment:
so.  in general, i'm pretty concerned about this sort of duplication happening all over the place.  It's not at all clear, IMO, for features to know tey need to do this sort of this.  It's also not clear why this wouldn't just be specified on the *option* itself.  Then the option could just be used to create all this data.

Reply:
still have this concern.